### ISSUE TITLE
New lints: Advisory lints around changes in trait sealing status

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#1119]
Turning an unsealed trait into a public API sealed (or unconditionally sealed) trait is a major breaking change, and we already have a lint for that. But other changes to the sealing status of a trait are possibly high-risk changes — they can be difficult to spot and may be difficult to revert (e.g. requiring a major breaking change). We should offer warning-level lints for those cases. For now, we can make these lints require "minor" SemVer versions — eventually, we'll have a way to make lints like this run on all versions regardless of SemVer bump (#1124).

- [x] #1120
- [x] #1121
- [x] #1122
- [x] #1123

**Comment by obi1kenobi** [comment]
Go for it!

**Comment by kernel-loophole** [comment]
hi @obi1kenobi i have a question regarding this issue ,What specific changes in trait sealing status should trigger the lints?

**Comment by obi1kenobi** [comment]
There's a list of suggested changes to look for in the issue description.
Each suggested new lint is a linked sub-issue. Make sure to read it
thoroughly, together with the contributing document and especially its
section on writing new lints.

Unfortunately, there's no way to run one lint at a time. If you really
wanted to, you could direct cargo to only run a specific test by name. But
generally, we've found that the approach in the contributing doc is easiest
and most successful.

**Comment by kernel-loophole** [comment]
okay great

**Comment by kernel-loophole** [comment]
@obi1kenobi here is the implementation of trait sealing lint of https://github.com/obi1kenobi/cargo-semver-checks/commit/8637f90a1391747fc245fdbc44d7ef3bc310ed04#diff-d743e40f010c5552b5f9b812d6888686b077adff06ca8e5f5162ac9f20e630d4 .i am looking for some example crate to test .i follow the adding lint guidelines.can you explain bit about this step 
```
span_: span @optional {
  filename @output
  begin_line @output
  end_line @output
}
```

**Comment by obi1kenobi** [comment]
I'm sorry, but the linked implementation is ignoring the explicit guidance of the contributing doc, and is of such low quality that I'm convinced it was entirely LLM-generated.

Please do not do this. If we wanted an LLM to be a GSoC contributor, we would have just used it ourselves. I'd be surprised if there is any GSoC project that would accept purely LLM-generated contributions, and it definitely isn't this one. In the interest of safeguarding my time for the Rust community's benefit, I'm not able to offer more advice than this, nor review more of your code.

**Comment by kernel-loophole** [comment]
@obi1kenobi that totally true .i wanted to use this as starting point .this is not like the final version or i have Make a PR of it .the reason i make request to review this is that i wanted to know if i am in right direction .

**Comment by obi1kenobi** [comment]
All lints merged!
